pyproject.toml

```diff
@@ -53,24 +53,25 @@ gitpython = "^3.1.18"
 pandas = ">=1.1.5"
 passlib = { extras = ["bcrypt"], version = "~1.7.4" }
 psutil = ">=5.0.0"
-pydantic = "<1.11, >=1.9.0"
+pydantic = "~2.7"
+pydantic-settings = "~2.2.1"
 pymysql = { version = "~1.0.2" }
 pyparsing = "<3,>=2.4.0"
 python = ">=3.8,<3.12"
 python-dateutil = "^2.8.1"
 pyyaml = ">=6.0.1"
 rich = { extras = ["jupyter"], version = ">=12.0.0" }
 secure = "~0.3.0"
-sqlalchemy_utils = "0.38.3"
-sqlmodel = "0.0.8"
+sqlalchemy = "^2.0.0"
+sqlalchemy_utils = "*"
```

Comment:
Are you talking about `sqlalchemy` or the utils here? I am a bit in between here. For both of them, I will try pinning it to a singular version and will execute the cross-compatibility checks again. If it works without any issues, it might be nice to pin them.